### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [hch12907/rfcmei#1]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install` or building the Release Docker image, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I have made quick tests (Fedora 41) by adding `lto = true` to the Release profile. The binary size reduction is from 3.6 Mib to 2.7 Mib.

I also highly recommend you enable LTO in the same way for your other Rust projects like https://github.com/hch12907/v6synth-dns - here the binary size reduction is from 14 Mib to 11 Mib.

Thank you.

P.S. By the way, the current `rfcmei` version cannot be compiled with the stable Rust 1.82. When I tried to invoke `cargo build --release`, I got this error:

```
   Compiling rfcmei v0.1.0 (/home/zamazan4ik/open_source/rfcmei)
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/main.rs:1:1
  |
1 | #![feature(let_chains)]
  | ^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
error: could not compile `rfcmei` (bin "rfcmei") due to 1 previous error
```

I suppose that at least for now the build instruction should be fixed to `cargo +nightly build --release` or to something similar.

**Comment by hch12907** [comment]
This is a good suggestion. I will update the release profile in Cargo.toml and also change the README a bit.

**Comment by hch12907** [comment]
Closed by ec2dfc3523bd84cc4701d0a601a9bca897fa2795
